bridges/snowbridge/pallets/outbound-queue-v2/src/lib.rs

```diff
@@ -358,7 +383,14 @@ pub mod pallet {
 			};
 			<PendingOrders<T>>::insert(nonce, order);
 
-			Nonce::<T>::set(nonce.checked_add(1).ok_or(Unsupported)?);
+			Nonce::<T>::set(nonce.checked_add(1).ok_or_else(|| {
+				Self::deposit_event(Event::MessageRejected {
+					id: Some(id),
+					nonce: Some(nonce),
+					error: Unsupported,
+				});
+				Unsupported
+			})?);
```

Comment:
This seems unnessessary, prefer be more simple something like:

```
Nonce::<T>::set(nonce.checked_add(1).expect("don't expect live long enough for 2^32 entries"));
```

Reply:
I prefer having the event. You never know, perhaps with massive increase in usage this could be reached 😉